lib/node_modules/@stdlib/assert/deep-has-own-property/test/test.js

```diff
@@ -0,0 +1,254 @@
+
+	throwsErr = incorrectSepOptionValue( null );
+	test.throws( throwsErr, TypeError, 'throws error when provided null as a sep option' );
+
+	throwsErr = incorrectSepOptionValue( void 0 );
+	test.throws( throwsErr, TypeError, 'throws error when provided undefined as a sep option' );
+
+	throwsErr = incorrectSepOptionValue( NaN );
+	test.throws( throwsErr, TypeError, 'throws error when provided NaN as a sep option' );
+
+	throwsErr = incorrectSepOptionValue( 3 );
+	test.throws( throwsErr, TypeError, 'throws error when provided number as a sep option' );
+
+	throwsErr = incorrectSepOptionValue( [] );
+	test.throws( throwsErr, TypeError, 'throws error when provided array as a sep option' );
+
+	test.end();
+
+	function incorrectSepOptionValue( sepValue ) {
+		return function incorrectSepOptionValue() {
+			var obj = {
+				'a': {
+					'b': 'c'
+				}
+			};
+			deepHasOwnProp( obj, 'a.b', {
+				'sep': sepValue
+			});
+		};
+	}
+});
+
+tape( 'the function returns `true` if an object has a specified path (defined as a delimited string)', function test( test ) {
+	var bool;
+	var dHasOwn;
+
+	bool = deepHasOwnProp( { 'a': { 'b': { 'c': 'd' } } }, 'a.b.c' );
+	test.strictEqual( bool, true, 'returns true' );
+
+	bool = deepHasOwnProp( { 'a': [ {'b': [ { 'c': 'd' }, { 'e': 'f' } ] } ] }, '0.a.0.b.0.c' );
+	test.strictEqual( bool, true, 'returns true' );
+
+	dHasOwn = deepHasOwnProp.factory( 'a.b.c' );
+	bool = 	dHasOwn( { 'a': { 'b': { 'c': 'd' } } } );
+	test.strictEqual( bool, true, 'returns true' );
+
+	test.end();
+});
+
+tape( 'the function returns `true` if an object has a specified path (defined as a key array)', function test( test ) {
+	var bool;
+	var dHasOwn;
+
+	bool = deepHasOwnProp( { 'a': { 'b': { 'c': 'd' } } }, [ 'a', 'b', 'c' ] );
+	test.strictEqual( bool, true, 'returns true' );
+
+	bool = deepHasOwnProp( { 'a': [ {'b': [ { 'c': 'd' }, { 'e': 'f' } ] } ] }, [ '0', 'a', '0', 'b', '0', 'c' ] );
+	test.strictEqual( bool, true, 'returns true' );
+
+	dHasOwn = deepHasOwnProp.factory([ 'a', 'b', '0', 'c' ]);
+	bool = dHasOwn( { 'a': { 'b': [ { 'c': 'd' } ] } } );
+	test.strictEqual( bool, true, 'returns true' );
+
+	test.end();
+});
+
+tape( 'the function returns `true` if an object has a specified path (using non-default delimiter)', function test( test ) {
+	var bool;
+	var dHasOwn;
+
+	bool = deepHasOwnProp( { 'a': { 'b': { 'c': 'd' } } }, 'a/b/c', {
+		'sep': '/'
+	});
+	test.strictEqual( bool, true, 'returns true' );
+
+	bool = deepHasOwnProp( { 'a': [ {'b': [ { 'c': 'd' }, { 'e': 'f' } ] } ] }, '0/a/0/b/0/c', {
+		'sep': '/'
+	});
+	test.strictEqual( bool, true, 'returns true' );
+
+	dHasOwn = deepHasOwnProp.factory('a/b/0/c', {
+		'sep': '/'
+	});
+	bool = dHasOwn( { 'a': { 'b': [ { 'c': 'd' } ] } } );
+	test.strictEqual( bool, true, 'returns true' );
+
+	test.end();
+});
+
+tape( 'the function returns `false` if an object does not have a specified path (defined as a delimited string)', function test( test ) {
+	var bool;
```

Comment:
Variable order and name. Also, for the main export tests, we do not need to test the `factory` method. You have already tested the `factory` method in the file dedicated to factory tests. I would remove all `factory` related tests from this file except for asserting that the main export has a `factory` method.